Source/Particles/Pusher/GetAndSetPosition.H

```diff
@@ -51,20 +52,49 @@ struct GetParticlePosition
      *         `x`, `y`, `z` */
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
     void operator() (const int i, RType& x, RType& y, RType& z) const noexcept
+    {
+        unpack_particle(m_structs[i], x, y, z);
+    }
+
+    /** \brief Extract the cartesian position coordinates of the particle
+    *         p and store them in the variables `x`, `y`, `z`
+    *         Particle version. */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void operator() (const PType& p, RType& x, RType& y, RType& z) const noexcept
+    {
+        unpack_particle(p, x, y, z);
+    }
+
+    /** \brief Extract the cartesian position coordinates of the particle
+    *         p and store them in the variables `x`, `y`, `z`
+    *         SuperParticle version. */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void operator() (const SuperPType& p, RType& x, RType& y, RType& z) const noexcept
+    {
+        unpack_particle(p, x, y, z);
+    }
+
+private:
+    /** \brief Extract the cartesian position coordinates of the particle
+    *         p and store them in the variables `x`, `y`, `z`
+    *         This is the implementaton and can't be called directly */
+    template <typename P>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void unpack_particle (const P& p, RType& x, RType& y, RType& z) const noexcept
```

Comment:
Offline discussed: we move this out to a free-standing `get_particle_position` function
```C++
template <typename P>
AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
void unpack_particle_position (const P& p, const RType theta,
                               RType& x, RType& y, RType& z) const noexcept
```